### PR TITLE
[WIP] ToParams() reflection-free experiment

### PR DESCRIPTION
**Pull request by matejvasek**
Attempt to make `ToParam()` method reflection-free by code-generation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -473,7 +473,7 @@ podman-remote-%-release:
 .PHONY: generate-bindings
 generate-bindings:
 ifneq ($(shell uname -s), Darwin)
-	GO111MODULE=off $(GO) generate ./pkg/bindings/... ;
+	GO111MODULE=on $(GO) generate ./pkg/bindings/... ;
```
Comment (matejvasek) [inline Makefile:476]: @vrothberg why was this changed to `off` before?

File: pkg/bindings/containers/types_execstartandattach_options.go [hunk 2]
```diff
@@ -26,53 +22,35 @@ func (o *ExecStartAndAttachOptions) Changed(fieldName string) bool {
...
+
 	if o == nil {
 		return params, nil
 	}
-	json := jsoniter.ConfigCompatibleWithStandardLibrary
-	s := reflect.ValueOf(o)
-	if reflect.Ptr == s.Kind() {
-		s = s.Elem()
+
+	if o.OutputStream != nil {
```
Comment (matejvasek) [inline pkg/bindings/containers/types_execstartandattach_options.go:30]: Those options structs are supposed to be send over wire, right? How can one serialize `io.Writer/Reader`?

File: pkg/bindings/images/types_import_options.go [hunk 3]
```diff
@@ -24,53 +19,27 @@ func (o *ImportOptions) Changed(fieldName string) bool {
...
 	if o == nil {
 		return params, nil
 	}
-	json := jsoniter.ConfigCompatibleWithStandardLibrary
-	s := reflect.ValueOf(o)
-	if reflect.Ptr == s.Kind() {
-		s = s.Elem()
+
+	if o.Changes != nil {
+		panic("*** GENERATOR DOESN'T IMPLEMENT THIS YET ***")
```
Comment (matejvasek) [inline pkg/bindings/images/types_import_options.go:28]: This crashes one of tests, however I believe it wasn't functioning correctly even before. `o.Changes` is `map[string]string` and therefore it was ignored before.

```go
lowerCaseKeys[iter.Key().Interface().(string)] = iter.Value().Interface().([]string)
```
If I understand it correcty `iter.Value().Interface().([]string)` will result in null if values happens to be just `string` instead of slice of strings.

Comment (matejvasek) [reply pkg/bindings/images/types_import_options.go:28]: Oh I see `o.Changes` is not `map[string]string`, it's `*[]string`! I don't see any reason to use `*[]string` since `[]string` is already nullable, weird.

Comment (matejvasek) [reply pkg/bindings/images/types_import_options.go:28]: The original code is removing pointer:
```golang
		if reflect.Ptr == f.Kind() {
			f = f.Elem()
		}
```

File: pkg/bindings/containers/types_execstartandattach_options.go (hunk 2 above)
Comment (Luap99) [reply pkg/bindings/containers/types_execstartandattach_options.go:30]: They are not supposed to be send over wire. see: https://github.com/containers/podman/blob/afe4ce6b1ce6a5ba21472f51defd243bdd7750b3/pkg/bindings/containers/attach.go#L500-L519

Comment (matejvasek) [reply pkg/bindings/containers/types_execstartandattach_options.go:30]: @Luap99  So only some struct fields (structurally simple ones) are serialized. The other may be handled differently (e.g. post as JSON in body). How it's decided? Is it solely based on type. For instance `int` fields are include in uri params, but struct/interface are not?

File: pkg/bindings/images/types_import_options.go (hunk 3 above)
Comment (matejvasek) [reply pkg/bindings/images/types_import_options.go]: Why is `map[string][]string` supported but `map[string]string` isn't?

File: pkg/bindings/containers/types_execstartandattach_options.go (hunk 2 above)
Comment (Luap99) [reply pkg/bindings/containers/types_execstartandattach_options.go:30]: I have no idea. @baude wrote the bindings stuff

**Comment by rhatdan** [comment]
Since the other PR is going in, is this still needed?

**Comment by rhatdan** [comment]
@matejvasek What is happening with this PR?  Is this still something you are working on? Or should we close it?
